File: Models/ViewModels/InvoiceVM.cs
using GoWheels_WebAPI.Models.Entities;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.Text.Json.Serialization;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class InvoiceVM : BaseModelVM
    {
        public decimal PrePayment { get; set; }
        public decimal Total { get; set; }
        public bool IsPay {  get; set; }
        public bool RefundInvoice { get; set; }
        public DateTime ReturnOn { get; set; }
        [JsonPropertyOrder(-100)]
        public BookingVM Booking { get; set; } = null!;
        public DriverBookingVM DriverBooking { get; set; } = null!;
    }
}

File: FE/src/Model/InvoiceVM.js
// src/models/InvoiceVM.js

import BookingVM from './BookingVM';
import DriverBookingVM from './DriverBookingVM';
import BaseModel from './BaseModelVM';
export default class InvoiceVM extends BaseModel{
  constructor(data = {}) {
    super(data.id || null, data.createdById, data.createdOn, data.modifiedById, data.modifiedOn, data.isDeleted);
    this.total = data.total || 0;
    this.returnOn = data.returnOn || null; // Date
    this.refundInvoice = data.refundInvoice || null;
    this.booking = new BookingVM(data.booking || {});
    this.driverBooking = new DriverBookingVM(data.driverBooking || {});
  }
}
